addons/pos_self_order/static/src/app/components/product_card/product_card.js

```diff
@@ -25,36 +25,39 @@ export class ProductCard extends Component {
             return;
         }
 
-        const cartButton = document.querySelector(".cart");
-        if (!cartButton || window.getComputedStyle(cartButton).display === "none") {
+        const toOrder = document.querySelector(".to-order");
+        if (!toOrder || window.getComputedStyle(toOrder).display === "none") {
             return;
         }
 
-        const clonedCard = productCardEl.cloneNode(true);
-        const cardRect = productCardEl.getBoundingClientRect();
-        const cartRect = cartButton.getBoundingClientRect();
+        const pic = this.selfRef.el.querySelector(".o_self_order_item_card_image");
+        if (!pic) {
+            return;
+        }
```

Comment:
if I have a good understanding, when a product don't have an image, these isn't animation ? I'm OK with that, but maybe we will need to talk about with the PO. 

This will not prevent the merge.

Reply:
This condition is indeed not necessary since products always have a picture (if no pics is defined the system will fallback to the placeholder).